NorthstarDLL/modmanager.cpp

```diff
@@ -426,6 +426,42 @@ void ModManager::LoadMods()
 
 					modPak.m_sPakName = pakName;
 
+					// read header of file and get the starpak paths
+
+					std::ifstream rpakStream(file.path(), std::ios::binary);
+
+					// seek to the point in the header where the starpak reference size is
+					rpakStream.seekg(0x38, std::ios::beg);
+					uint16_t starpaksSize;
+					rpakStream.read((char*)&starpaksSize, 2);
+
+					// seek to just after the header
+					rpakStream.seekg(0x58, std::ios::beg);
+					// read the starpak reference(s)
+					std::vector<char> buf(starpaksSize);
+					rpakStream.read(buf.data(), starpaksSize);
+
+					rpakStream.close();
+
+					// split the starpak reference(s) into strings to hash
+					std::string str = "";
+					for (uint16_t i = 0; i < starpaksSize; i++)
+					{
+						if (buf[i] != 0x00)
+						{
+							str += buf[i];
+						}
+						else
+						{
+							if (str != "")
```

Comment:
!str.empty()

Reply:
Out of curiosity, what is the difference between the two?

Reply:
One uses compare (a branch), one checks the size field directly (a load). Minor difference, but it can add up.